src/lib/Libecl/pbs_client_thread.c

```diff
@@ -639,6 +644,10 @@ __pbs_client_thread_init_thread_context(void)
 	 * to read the error code out of it.
 	 */
 	pbs_client_thread_set_single_threaded_mode();
+	if (free_ptr) {
```

Comment:
I don't know much about the pthread stuff, but what I assuming what is happening is at line 617 we call pthread_set_specific() with ptr.  I'm assuming this associates ptr with a thread since we don't free ptr in the success case.  There are some times we goto err after the call to pthread_set_specific().  We can't just check if ptr is NULL, because we don't know if we got there before or after the call to pthread_set_specific().  You're more of an expert on pthreads than I am, so please let me know if I am wrong.

Reply:
Oh, my bad, I hadn't seen the call to pthread_setspecific. You are right, we should not be freeing it after pthread_setspecific call until we destroy the thread itself.